www/js/theme_editor.js

```diff
@@ -68,70 +52,61 @@ const swatchFactory = themeInstance =>
       },
       themeInstance.name));
 
-/*
-const bgTileFactory = (bgImage) => {
-  const bgImageUrl = bgImage ? `url(../${imagesPath}/${bgImage})` : '';
-  return h(
-    'li.theme-instance',
+const swatchHeaderFactory = headerText => h('header.options-header', headerText);
+
+const imageInput = () =>
+  h(
+    'label.file-input-label',
     {
-      style: {
-        'background-image': bgImageUrl,
-      },
-      onclick: () => {
-        const currentTheme = getCurrentTheme();
-        currentTheme.bgImage = bgImage;
-        try {
-          localStorage.setItem('customTheme', JSON.stringify(currentTheme));
-          global.core.platformMethod('updateTheme');
-        } catch (error) {
-          new Noty({
-            type: 'error',
-            text: `There is an error adding this background to current theme.
-            Try checking themes.json for errors. If error persists,
-            report it at www.sttm.co`,
-            timeout: 3000,
-            modal: true,
-          }).show();
-        }
-      },
+      for: 'themebg-upload',
     },
+    'Choose a file',
+    h('input.file-input#themebg-upload',
+      {
+        type: 'file',
+        accept: 'image/*',
+        onchange: (evt) => {
+          const curTheme = store.getUserPref('app.theme');
+          if (!themesWithCustomBg.includes(curTheme)) {
+            store.setUserPref('app.theme', themesWithCustomBg[0]);
+          }
+          store.setUserPref('app.themebg', {
+            type: 'custom',
+            url: evt.target.files[0].path.replace(/(\s)/g, '\\ '),
```

Comment:
Could we please run an optimizer on the image and make a copy of it in the userData directory and use that? https://www.npmjs.com/package/imagemin